src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, Signal, computed, inject, viewChild } from '@angular/core';
+import { LearningPathService } from 'app/course/learning-paths/learning-path.service';
+import { LearningObjectType } from 'app/entities/competency/learning-path.model';
+import { Observable, catchError, map, of, switchMap } from 'rxjs';
+import { onError } from 'app/shared/util/global.utils';
+import { HttpErrorResponse } from '@angular/common/http';
+import { toObservable, toSignal } from '@angular/core/rxjs-interop';
+import { CommonModule } from '@angular/common';
+import { LearningPathStudentNavComponent, LoadedValue } from 'app/course/learning-paths/components/learning-path-student-nav/learning-path-student-nav.component';
+import { AlertService } from 'app/core/util/alert.service';
+import { ActivatedRoute, RouterModule } from '@angular/router';
+import { CourseExerciseDetailsModule } from 'app/overview/exercise-details/course-exercise-details.module';
+import { LearningPathLectureUnitComponent } from 'app/course/learning-paths/components/learning-path-lecture-unit/learning-path-lecture-unit.component';
+import { LearningPathExerciseComponent } from 'app/course/learning-paths/components/learning-path-exercise/learning-path-exercise.component';
+
+@Component({
+    selector: 'jhi-learning-path-student-page',
+    templateUrl: './learning-path-student-page.component.html',
+    styleUrl: './learning-path-student-page.component.scss',
+    standalone: true,
+    imports: [CommonModule, RouterModule, LearningPathStudentNavComponent, CourseExerciseDetailsModule, LearningPathLectureUnitComponent, LearningPathExerciseComponent],
+})
+export class LearningPathStudentPageComponent {
+    protected readonly LearningObjectType = LearningObjectType;
+
+    private readonly learningPathService = inject(LearningPathService);
+    private readonly alertService = inject(AlertService);
+    private readonly activatedRoute = inject(ActivatedRoute);
+
+    private readonly navigation = viewChild(LearningPathStudentNavComponent);
+
+    readonly courseId: Signal<number> = toSignal(this.activatedRoute.parent!.parent!.params.pipe(map((params) => params.courseId)));
```

Comment:
The code currently provides a type declaration for some signals and no type declaration for other signals. Could we make this consistent?

The inline markers in IntellIJ can only be so helpful (the three learning object signals here  `Signal<LearningPathNavigationObjectDto | undefined>`):
![image](https://github.com/ls1intum/Artemis/assets/38403547/3f504568-ffa3-4230-9757-22d2172a16ce)

Reply:
In some cases it is needed to ensure that the type is specific and not any